checks/prometheus_check.py

```diff
@@ -284,12 +329,49 @@ def process_metric(self, message, send_histograms_buckets=True, custom_tags=None
         `send_histograms_buckets` is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
         """
         try:
+            # If targeted metric, store labels
+            if message.name in self.label_joins:
```

Comment:
We can also here drop the if statement with a KeyError:
```python
            try:
                matching_label = self.label_joins[message.name]['label_to_match']
                for metric in message.metric:
                    labels_list = []
                    matching_value = None
                    for label in metric.label:
                        if label.name == matching_label:
                            matching_value = label.value
                        elif label.name in self.label_joins[message.name]['labels_to_get']:
                            labels_list.append((label.name, label.value))
                    try:
                        self._label_mapping[matching_label][matching_value] = labels_list
                    except KeyError:
                        if matching_value is not None:
                            self._label_mapping[matching_label] = {matching_value: labels_list}
            except KeyError:
                pass
```

Reply:
I'll probably leave the `if` on this on I found the `try` block too big